internal/manifest/vmdk.go

```diff
@@ -53,3 +53,45 @@ func (p *VMDK) Export() *artifact.Artifact {
 	mimeType := "application/x-vmdk"
 	return artifact.New(p.Name(), p.Filename, &mimeType)
 }
+
+// A VMDK turns a raw ostree image file into vmdk image.
+type VMDKOstree struct {
+	Base
+	Filename string
+
+	imgPipeline *RawOSTreeImage
+}
+
+func NewVMDKFromOstreeRaw(m *Manifest, buildPipeline *Build, imgPipeline *RawOSTreeImage) *VMDKOstree {
```

Comment:
I would suggest to instead extend the `NewVMDK()` function to accept additional type of `imgPipeline`. Mainly because the rest of the code is basically identical at the first glance. The VMDK pipeline should not care necessarily which type of pipeline produces the raw image, it is just that we didn't have this use case before.

Reply:
The problem is that the `VMDK` struct takes a `RawImage` instead of a `RawOSTreeImage`, so I would have to add another field in the struct to differentiate between them, so making a different struct with the associated functions seemed cleaner. What do you think?

Reply:
You wouldn't need to add another struct, you could make the `imgPipeline` argument of type `Pipeline` which is the interface describing them *however* there is no `Filename` attribute or method in that interface so you indeed would end up with adding a different field, ensuring that not both are set, ensuring that at least one is set, etc.

Because that's (to me) more messy than duplicating these 30 lines of code. I'm ok with the current approach.

Reply:
Actually, the `Pipeline` interface does contain `Export() *artifact.Artifact` method, which contains the `Filename`.

Reply:
mmm I'll take another look